c7n/policy.py

```diff
@@ -1196,7 +1196,9 @@ def _write_file(self, rel_path, value):
 
     def load_resource_manager(self):
         factory = get_resource_class(self.data.get('resource'))
-        return factory(self.ctx, self.data)
+        resource_manager = factory(self.ctx, self.data)
+        clouds[self.provider_name]().initialize_resource(resource_manager)
```

Comment:
this hook point is too late to participate in schema validation, ie. its on instance creation time instead of class registration time.

Reply:
I'll take a look.

I thought that this method is called after expanding the variables, which should happen before schema validation:

https://github.com/cloud-custodian/cloud-custodian/blob/4a529478c009a491962215bf4e272ac035a2449f/c7n/policy.py#L1115-L1132

https://github.com/cloud-custodian/cloud-custodian/blob/4a529478c009a491962215bf4e272ac035a2449f/c7n/commands.py#L121-L123

Reply:
I think that I see the issue; I misunderstood that there was a difference between schema validate and non-schema validate.

Reply:
I moved this to https://github.com/cloud-custodian/cloud-custodian/pull/6504/files#diff-243a00e2507c3f91acfc00d5558a293736f2f38b90afc6c3b936ca68ac676c14R26